### ISSUE TITLE
Filtering by related models on eagerly loaded models

### ISSUE THREAD

**Issue by jeff3yan** [tandg-digital/objection-filter#8]
With some minor changes to the code base, it'll be possible to filter by related models not just on the root model - but also on any eagerly loaded models.

An example of this query is something like:
1. Give me all Customers where the customer's city begins with "A"
2. For those customers, eagerly load their pets, but only the pets where the breed begins with "C"

I'm thinking the query structure for this will look like:
```
{
  "eager": {
    "$where": {
      "city.name": {
        "$like": "A%"
      }
    },
    "pets": {
      "$where": {
        "breed.name": {
          "$like": "C%"
        }
      }
    }
  }
}
```

At the moment, inside a `$where` modifier, it only allows the use of column names on the target model. With this change, it'll allow you to use any related field of any eager model.

**Comment by jeff3yan** [comment]
Implemented in the [develop branch](https://github.com/tandg-digital/objection-filter/commit/eaa013ec90b64234a1fc330d8e0789c3faf9a64e). Once documentation is done will be merged into master.
